core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowResource.scala

```diff
@@ -563,4 +607,132 @@ class WorkflowResource {
     }
   }
 
+  /**
+    * Helper function to retrieve the owner filter.
+    * Applies a filter based on the specified owner names.
+    *
+    * @param owners The list of owner names to filter by.
+    * @return The owner filter.
+    */
+  def getOwnerFilter(owners: java.util.List[String]): Condition = {
+    var ownerFilter: Condition = noCondition()
+    if (owners != null && !owners.isEmpty) {
+      for (owner <- owners) {
+        ownerFilter = ownerFilter.or(USER.NAME.eq(owner))
+      }
+    }
+    ownerFilter
+  }
+
+  /**
+    * Helper function to retrieve the project filter.
+    * Applies a filter based on the specified project IDs.
+    *
+    * @param projectIds The list of owner names to filter by.
+    * @return The projectId filter.
+    */
+  def getProjectFilter(projectIds: java.util.List[UInteger]): Condition = {
+    var projectIdFilter: Condition = noCondition()
+    if (projectIds.nonEmpty) {
+      for (projectId <- projectIds) {
+        projectIdFilter = projectIdFilter.or(WORKFLOW_OF_PROJECT.PID.eq(projectId))
+      }
+    }
+    projectIdFilter
+  }
+
+  /**
+    * Helper function to retrieve the workflowID filter.
+    * Applies a filter based on the specified workflow IDs.
+    * @param workflowIDs The list of workflow IDs to filter by.
+    * @return The workflowID filter.
+    */
+  def getWorkflowIdFilter(workflowIDs: java.util.List[UInteger]): Condition = {
+    var workflowIdFilter: Condition = noCondition()
+    if (workflowIDs != null && !workflowIDs.isEmpty) {
+      for (workflowID <- workflowIDs) {
+        workflowIdFilter = workflowIdFilter.or(WORKFLOW.WID.eq(workflowID))
+      }
+    }
+    workflowIdFilter
+  }
+
+  /**
+    * Helper function to retrieve the creation date filter.
+    * Applies a filter based on the specified creation start and end dates.
+    *
+    * @param creationStartDate The start date to filter by.
+    * @param creationEndDate   The end date to filter by.
+    * @return The creation date filter.
+    */
+  def getCreationDateFilter(
+      creationStartDate: String,
```

Comment:
This function can be consolidated with getModifiedDateFilter into maybe a getDateFilter function. WORKFLOW.CREATION_TIME and WORKFLOW.LAST_MODIFIED_TIME can be passed as arguments.

Reply:
done